[BUG] Unable to run opensearch-benchmark in test mode

**Describe the bug**
- opensearch-benchmark fails without running the benchmark on a local OpenSearch server
- Logs below

**To Reproduce**
1. Install latest version from pypi (pip3 install opensearch-benchmark)
2. Execute the test command
```
opensearch-benchmark execute_test --target-host=localhost:9200 --workload=nyc_taxis --pipeline=benchmark-only --test-mode --kill-running-processes
```

**Expected behavior**
- Benchmarks to run in test mode

**Logs**
```
023-03-27 23:25:05,311 ActorAddr-(T|:49871)/PID:73989 osbenchmark.actor ERROR Error in test execution orchestrator
Traceback (most recent call last):

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/actor.py", line 92, in guard
    return f(self, msg, sender)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/test_execution_orchestrator.py", line 108, in receiveMsg_Setup
    self.coordinator.setup(sources=msg.sources)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/test_execution_orchestrator.py", line 195, in setup
    self.current_workload = workload.load_workload(self.cfg)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/workload/loader.py", line 192, in load_workload
    repo = workload_repo(cfg)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/workload/loader.py", line 290, in workload_repo
    return GitWorkloadRepository(cfg, fetch, update)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/workload/loader.py", line 331, in __init__
    self.repo.update(distribution_version)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/utils/repo.py", line 68, in update
    branch = versions.best_match(git.branches(self.repo_dir, remote=self.remote), distribution_version)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/utils/git.py", line 44, in probe
    return f(src, *args, **kwargs)

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/utils/git.py", line 123, in branches
    return _cleanup_remote_branch_names(process.run_subprocess_with_output(

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/utils/git.py", line 137, in _cleanup_remote_branch_names
    return [(b[b.index("/") + 1:]).strip() for b in branch_names if not b.endswith("/HEAD")]

  File "/Users/kkotwani/.pyenv/versions/3.8.16/lib/python3.8/site-packages/osbenchmark/utils/git.py", line 137, in <listcomp>
    return [(b[b.index("/") + 1:]).strip() for b in branch_names if not b.endswith("/HEAD")]

ValueError: substring not found
```

**More Context (please complete the following information):**
- Workload(Share link for custom workloads): nyc_taxis
- Service(E.g OpenSearch): OpenSearch
- Version (E.g. 1.0): 3.0.0/Latest main

**Additional context**
- Running on an arm64 M1 Macbook

@kartg Thanks for diving into this! Just caught up on the thread. TL;DR: Reproduced the issue in a clean Ubuntu environment and can confirm that the issue lies in `Git` versions 2.40.0+. Propose we use @gkamat recommended fix after testing them out. 

I'm in the same boat as Govind and am using an earlier version of `Git`, which doesn't exclude the `HEAD` from `origin/HEAD`:
```
hoangia@3c22fbd0d988 default % git for-each-ref refs/remotes/ --format='%(refname:short)'
origin/1
origin/2
origin/3
origin/6
origin/7
origin/HEAD
origin/main
hoangia@3c22fbd0d988 default % git --version
git version 2.33.0
```

### Reproduced issue in Ubuntu:
Confirmed that issue resides in `git` versioning. 
1. Started with git version 2.33.0
```
ubuntu@ip-172-31-80-80:~$ opensearch-benchmark list workloads

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

Available workloads:

Name           Description                                                                                                        Documents    Compressed Size    Uncompressed Size    Default TestProcedure         All TestProcedures
-------------  -----------------------------------------------------------------------------------------------------------------  -----------  -----------------  -------------------  ----------------------------  ------------------------------------------------------------------------------------------------------------------------------------------------------------------
pmc            Full text benchmark with academic papers from PMC                                                                  574,199      5.5 GB             21.7 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-fast-with-conflicts
nested         StackOverflow Q&A stored as nested docs                                                                            11,203,029   663.3 MB           3.4 GB               nested-search-test-procedure  nested-search-test-procedure,index-only
geoshape       Shapes from PlanetOSM                                                                                              60,523,283   13.4 GB            45.4 GB              append-no-conflicts           append-no-conflicts
percolator     Percolator benchmark based on AOL queries                                                                          2,000,000    121.1 kB           104.9 MB             append-no-conflicts           append-no-conflicts
so             Indexing benchmark using up to questions and answers from StackOverflow                                            36,062,278   8.9 GB             33.1 GB              append-no-conflicts           append-no-conflicts
noaa           Global daily weather measurements from NOAA                                                                        33,659,481   949.4 MB           9.0 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,top_metrics,aggs
http_logs      HTTP server log data                                                                                               247,249,096  1.2 GB             31.1 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-index-only-with-ingest-pipeline,update,append-no-conflicts-index-reindex-only
geopointshape  Point coordinates from PlanetOSM indexed as geoshapes                                                              60,844,404   470.8 MB           2.6 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-fast-with-conflicts
geopoint       Point coordinates from PlanetOSM                                                                                   60,844,404   482.1 MB           2.3 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-fast-with-conflicts
geonames       POIs from Geonames                                                                                                 11,396,503   252.9 MB           3.3 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-fast-with-conflicts,significant-text
nyc_taxis      Taxi rides in New York in 2015                                                                                     165,346,692  4.5 GB             74.3 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts-index-only,update,searchable-snapshot
eventdata      This benchmark indexes HTTP access logs generated based sample logs from the elastic.co website using a generator  20,000,000   756.0 MB           15.3 GB              append-no-conflicts           append-no-conflicts,transform

-------------------------------
[INFO] SUCCESS (took 1 seconds)
-------------------------------
ubuntu@ip-172-31-80-80:~$ git version
git version 2.34.1
```

2. Updated `git` to 2.40.0 and observed error when listing workloads
```
ubuntu@ip-172-31-80-80:~/.benchmark/benchmarks/workloads/default$ git --version
git version 2.40.0
ubuntu@ip-172-31-80-80:~/.benchmark/benchmarks/workloads/default$ opensearch-benchmark list workloads

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[ERROR] Cannot list. substring not found.
```

### Applying the recommended fix by @gkamat 
Added recommended fix by @gkamat to `opensearch-benchmark/osbenchmark/utils/git.py` and ran `list` subcommand for environments with `git` versions with `2.40.0` and `2.33.0`. 
1. Remove `:short` from `refname:short` in this line:
https://github.com/IanHoang/opensearch-benchmark/blob/a1f45502b5b9e69bad1c8d10e7a6c30bd0ed8469/osbenchmark/utils/git.py#L123-L124

2. Added `rindex` to `index` in this line: 
https://github.com/IanHoang/opensearch-benchmark/blob/a1f45502b5b9e69bad1c8d10e7a6c30bd0ed8469/osbenchmark/utils/git.py#L136-L137

3. Reran `python3 -m pip install -e .` to reinstall OSB in development mode. 
4. Reran `opensearch-benchmark list workloads` and received the successful output:
```
ubuntu@ip-172-31-80-80:~/opensearch-benchmark$ opensearch-benchmark list workloads

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

Available workloads:

Name           Description                                                                                                        Documents    Compressed Size    Uncompressed Size    Default TestProcedure         All TestProcedures
-------------  -----------------------------------------------------------------------------------------------------------------  -----------  -----------------  -------------------  ----------------------------  ------------------------------------------------------------------------------------------------------------------------------------------------------------------
pmc            Full text benchmark with academic papers from PMC                                                                  574,199      5.5 GB             21.7 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-fast-with-conflicts
nested         StackOverflow Q&A stored as nested docs                                                                            11,203,029   663.3 MB           3.4 GB               nested-search-test-procedure  nested-search-test-procedure,index-only
geoshape       Shapes from PlanetOSM                                                                                              60,523,283   13.4 GB            45.4 GB              append-no-conflicts           append-no-conflicts
percolator     Percolator benchmark based on AOL queries                                                                          2,000,000    121.1 kB           104.9 MB             append-no-conflicts           append-no-conflicts
so             Indexing benchmark using up to questions and answers from StackOverflow                                            36,062,278   8.9 GB             33.1 GB              append-no-conflicts           append-no-conflicts
noaa           Global daily weather measurements from NOAA                                                                        33,659,481   949.4 MB           9.0 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,top_metrics,aggs
http_logs      HTTP server log data                                                                                               247,249,096  1.2 GB             31.1 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-index-only-with-ingest-pipeline,update,append-no-conflicts-index-reindex-only
geopointshape  Point coordinates from PlanetOSM indexed as geoshapes                                                              60,844,404   470.8 MB           2.6 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-fast-with-conflicts
geopoint       Point coordinates from PlanetOSM                                                                                   60,844,404   482.1 MB           2.3 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-fast-with-conflicts
geonames       POIs from Geonames                                                                                                 11,396,503   252.9 MB           3.3 GB               append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts,append-fast-with-conflicts,significant-text
nyc_taxis      Taxi rides in New York in 2015                                                                                     165,346,692  4.5 GB             74.3 GB              append-no-conflicts           append-no-conflicts,append-no-conflicts-index-only,append-sorted-no-conflicts-index-only,update,searchable-snapshot
eventdata      This benchmark indexes HTTP access logs generated based sample logs from the elastic.co website using a generator  20,000,000   756.0 MB           15.3 GB              append-no-conflicts           append-no-conflicts,transform

-------------------------------
[INFO] SUCCESS (took 0 seconds)
-------------------------------
```

--- a/osbenchmark/utils/git.py
+++ b/osbenchmark/utils/git.py
@@ -121,7 +121,7 @@ def branches(src_dir, remote=True):
     if remote:
         # alternatively: git for-each-ref refs/remotes/ --format='%(refname:short)'
         return _cleanup_remote_branch_names(process.run_subprocess_with_output(
-                "git -C {src} for-each-ref refs/remotes/ --format='%(refname:short)'".format(src=clean_src)))
+                "git -C {src} for-each-ref refs/remotes/ --format='%(refname)'".format(src=clean_src)))
     else:
         return _cleanup_local_branch_names(
                 process.run_subprocess_with_output(
@@ -134,7 +134,7 @@ def tags(src_dir):
 
 
 def _cleanup_remote_branch_names(branch_names):
-    return [(b[b.index("/") + 1:]).strip() for b in branch_names if not b.endswith("/HEAD")]
+    return [(b[b.rindex("/") + 1:]).strip() for b in branch_names if not b.endswith("/HEAD")]
 
 
 def _cleanup_local_branch_names(branch_names):

--- a/tests/utils/git_test.py
+++ b/tests/utils/git_test.py
@@ -179,7 +179,7 @@ class GitTests(TestCase):
                                        "  origin/5.0.0-alpha1",
                                        "  origin/5"]
         self.assertEqual(["main", "5.0.0-alpha1", "5"], git.branches("/src", remote=True))
-        run_subprocess.assert_called_with("git -C /src for-each-ref refs/remotes/ --format='%(refname:short)'")
+        run_subprocess.assert_called_with("git -C /src for-each-ref refs/remotes/ --format='%(refname)'")
 
     @mock.patch("osbenchmark.utils.process.run_subprocess_with_output")
     @mock.patch("osbenchmark.utils.process.run_subprocess_with_logging")

